File: 0415-add-strings/0415-add-strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        res = collections.deque([])
        carry = 0
        
        i = len(num1) - 1
        j = len(num2) - 1
        
        while i >= 0 or j >= 0:
            cur_i = int(num1[i]) if i >= 0 else 0
            cur_j = int(num2[j]) if j >= 0 else 0
            
            cur_sum = carry +  cur_i + cur_j
            res.appendleft(str(cur_sum % 10))
            carry = cur_sum // 10
            
            i -= 1
            j -= 1
        
        if carry:
            res.appendleft(str(carry))
        
        return "".join(res)
            
        

File: 0415-add-strings/0415-add-strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
    let i1 = num1.length - 1;
    let i2 = num2.length - 1;
    let res = [];
    let carry = 0;
    
    while (i1 >= 0 || i2 >= 0) {
        let val1 = i1 >= 0 ? parseInt(num1[i1]) : 0;
        let val2 = i2 >= 0 ? parseInt(num2[i2]) : 0;
        
        let sum = val1 + val2 + carry;
        
        res.push(sum % 10);
        carry = Math.floor(sum / 10);
        i1--;
        i2--;
    }
    
    if (carry) {
        res.push(carry);
    }
    
    return res.reverse().join("")
    
};